### PR TITLE
feat(pull-request): Add support for running on forked repos

### PR DESCRIPTION
**Pull request by robinjoseph08**
### What

Add support for running this action and generating PR diagrams for forked PRs.

- When checking for if it's a `pull_request` event, also check for `pull_request_target` since that's what we'll be moving to. And even though we'll stop using `pull_request` moving forward, we'll still support it for backwards compatibility.
- Remove the preflight check that made sure that the head repo and the base repo are the same. This was explicitly preventing forked repos from running, so since we now want them to run, this needed to be removed.
- For a forked repo, the `origin` remote that is set up is the forked repo, not the base repo. So since we used that to determine the repo that we needed to upload the dependency map for, it was using the wrong repo, resulting in a 404. Now, we'll just be using the value of the `GITHUB_REPOSITORY` environment variable. This is the correct value for both branch PRs and fork PRs. I also confirmed that this env is set for `pull_request` events as well.
- Not entirely sure why there are so many changes to `dist/`. My guess is in how `ncc` ran, but I don't know enough of it to determine the root cause. I've run this against PRs though and have confirmed that it runs, so it should be fine.

### CODE REVIEW

**Review by deammer — APPROVED** [review]
👏 fantastic, this is very clean! I recommend waiting for a second approval as I'm not super familiar with this action code.

**Review by graysonarts — APPROVED** [review]
Ditto what @deammer said. and yay for simplifying code!

**Pull request merged**
